smartcontract/service/native/global_params/global_params.go

```diff
@@ -124,15 +130,29 @@ func TransferAdmin(native *native.NativeService) ([]byte, error) {
 	if err := destinationAdmin.Deserialize(bytes.NewBuffer(native.Input)); err != nil {
 		return utils.BYTE_FALSE, errors.NewErr("transfer admin, deserialize admin failed!")
 	}
-	native.CloneCache.Add(scommon.ST_STORAGE, GenerateAdminKey(contract, true),
+	native.CloneCache.Add(scommon.ST_STORAGE, generateAdminKey(contract, true),
 		getAdminStorageItem(destinationAdmin))
 	return utils.BYTE_TRUE, nil
 }
 
-func SetGlobalParam(native *native.NativeService) ([]byte, error) {
+func SetOperator(native *native.NativeService) ([]byte, error) {
 	contract := native.ContextRef.CurrentContext().ContractAddress
 	getAdmin(native, contract)
 	if !native.ContextRef.CheckWitness(common.Address(*admin)) {
+		return utils.BYTE_FALSE, errors.NewErr("set operator, authentication failed!")
+	}
+	destinationOperator := new(Admin)
+	if err := destinationOperator.Deserialize(bytes.NewBuffer(native.Input)); err != nil {
+		return utils.BYTE_FALSE, errors.NewErr("set operator, deserialize admin failed!")
+	}
+	native.CloneCache.Add(scommon.ST_STORAGE, GenerateOperatorKey(contract), getAdminStorageItem(destinationOperator))
```

Comment:
and `getAdminStorageItem` to `getRoleStorageItem`

Reply:
you forget to set global variable `operator`.  But I advise you to remove the global variable `admin` and `operator`, since global variable will effect parallel transaction execution.